### PR TITLE
677 modules and hooks folder

### PR DESCRIPTION
**Pull request by deekshithepns**
Added types wherever possible

### CODE REVIEW

**Review by corlard3y — COMMENTED** [review]
Conflicts need to be resolved, before it can be approved

**Comment by deekshithepns** [comment]
> Conflicts need to be resolved, before it can be approved

ok..I will fix that

**Comment by deekshithepns** [comment]
> Conflicts need to be resolved, before it can be approved

pushed changes

**Review by mishramonalisha76 — APPROVED** [review]

File: src/modules/nft/NftModule.tsx [hunk 1]
```diff
@@ -47,11 +47,11 @@ function NftModule() {
   }, [account]);
 
   // handle user action at control center
-  const userClickedAt = (controlIndex) => {
+  const userClickedAt = (controlIndex: number): void => {
     setVersion(controlIndex);
   };
 
-  const handleChange = (e) => {
+  const handleChange = (e: ChangeEvent<HTMLInputElement>): void => {
```
Comment (mishramonalisha76) [inline src/modules/nft/NftModule.tsx:54]: there is an error in ChangeEvent type

File: src/modules/yield/YieldFarmingModule.tsx [hunk 2]
```diff
@@ -17,87 +17,87 @@ import { abis, addresses, appConfig } from 'config';
...
   };
 
-  const getFormattedDuration = () => {
+  const getFormattedDuration = (): void => {
     if (poolStats?.epochEndTimestamp) {
-      const epochEndTimestamp = poolStats.epochEndTimestamp.toNumber();
+      const epochEndTimestamp: number = poolStats.epochEndTimestamp.toNumber();
 
-      const duration = epochEndTimestamp - Math.floor(new Date() / 1000);
+      const duration: number = epochEndTimestamp - Math.floor(new Date() / 1000);
```
Comment (mishramonalisha76) [inline src/modules/yield/YieldFarmingModule.tsx:90]: fix types errors in this function

**Review by mishramonalisha76 — CHANGES_REQUESTED** [review]

File: src/modules/inbox/InboxModule.tsx [hunk 3]
```diff
@@ -34,14 +34,14 @@ const InboxModule = () => {
 
   const dispatch = useDispatch();
   const { account, chainId, library } = useWeb3React();
-  const { epnsReadProvider, epnsCommReadProvider } = useSelector((state) => state.contracts);
+  const { epnsReadProvider, epnsCommReadProvider } = useSelector((state:any) => state.contracts);
```
Comment (mishramonalisha76) [inline src/modules/inbox/InboxModule.tsx:37]: image
make the input param like this, to resolve the error

**Review by mishramonalisha76 — CHANGES_REQUESTED** [review]

File: src/modules/inbox/InboxModule.tsx (hunk 3 above)
Comment (deekshithepns) [reply src/modules/inbox/InboxModule.tsx:37]: pushed all review changes...

**Comment by zaryab2000** [comment]
Has the conflicts not yet been resolved on this? @deekshithepns @corlard3y ?

**Comment by corlard3y** [comment]
@deekshithepns the conflicts can be fixed one last time, and if @mishramonalisha76 guarantees the code changes are fine, the PR can be merged

**Comment by deekshithepns** [comment]
> @deekshithepns the conflicts can be fixed one last time, and if @mishramonalisha76 guarantees the code changes are fine, the PR can be merged

resolved conflicts

**Comment by mishramonalisha76** [comment]
We will use the types in this pr for each file after code optimization. cc @0xNilesh

**Pull request closed without merging**